worker/restore_map.go

```diff
@@ -333,7 +340,11 @@ func (m *mapper) processReqCh(ctx context.Context) error {
 
 		// Update the local max uid and max namespace values.
 		maxUid = x.Max(maxUid, parsedKey.Uid)
-		maxNs = x.Max(maxNs, ns)
+		if in.oneNamespaceOnly {
```

Comment:
add a comment here, and I think this is not right. Try performing a namespace aware restore and then create an existing namesapce. It should not work.